pkgs/applications/networking/cluster/argocd/default.nix

```diff
@@ -1,40 +1,74 @@
-{ lib, buildGoModule, fetchFromGitHub, packr }:
+{ lib, buildGoModule, fetchFromGitHub, packr, makeWrapper, installShellFiles, helm, kustomize }:
 
 buildGoModule rec {
   pname = "argocd";
-  version = "1.8.6";
-  commit = "28aea3dfdede00443b52cc584814d80e8f896200";
+  version = "2.0.1";
+  commit = "33eaf11e3abd8c761c726e815cbb4b6af7dcb030";
+  tag = "v${version}";
 
   src = fetchFromGitHub {
     owner = "argoproj";
     repo = "argo-cd";
-    rev = "v${version}";
-    sha256 = "sha256-kJ3/1owK5T+FbcvjmK2CO+i/KwmVZRSGzF6fCt8J9E8=";
+    rev = tag;
+    sha256 = "sha256-j/RdiMeaYxlmEvo5CKrGvkp25MrFsSYh+XNYWNcs0PE=";
   };
 
-  vendorSha256 = "sha256-rZ/ox180h9scocheYtMmKkoHY2/jH+I++vYX8R0fdlA=";
+  vendorSha256 = "sha256-8j5v99wOHM/SndJwpmGWiCFEyw4K513HEEbkPrD8C90=";
 
+  nativeBuildInputs = [ packr makeWrapper installShellFiles ];
+
+  # run packr to embed assets
+  preBuild = ''
+    packr
+  '';
+
+  buildFlagsArray =
+    let package_url = "github.com/argoproj/argo-cd/v2/common"; in
+    [
+      "-ldflags="
+      "-s -w"
+      "-X ${package_url}.version=${version}"
+      "-X ${package_url}.buildDate=unknown"
+      "-X ${package_url}.gitCommit=${commit}"
+      "-X ${package_url}.gitTag=${tag}"
+      "-X ${package_url}.gitTreeState=clean"
+    ];
+
+  # Test is disabled because ksonnet is missing from nixpkgs.
+  # Log: https://gist.github.com/superherointj/79cbdc869dfd44d28a10dc6746ecb3f9
   doCheck = false;
+  checkInputs = [
+    helm
+    kustomize
+    #ksonnet
```

Comment:
```suggestion
```

Reply:
I'd prefer to leave it as a marker since today tests require ksonnet. In the future, when it is not required it may be removed.  I understand the package does not exist and may never exist. But I don't see it as noise but as relevant information to know what is going on.